engine: add known issues to 20.10.15 release notes

thaJeztah opened the pull request:
- relates to https://github.com/docker/cli/issues/3595
- relates to https://github.com/docker/cli/pull/3592

With this PR:

<img width="999" alt="Screenshot 2022-05-10 at 11 15 55" src="https://user-images.githubusercontent.com/1804568/167594494-087af734-29a3-4c41-b82e-53d4ab947661.png">

thaJeztah commented:
@usha-mandya @dockertopia ptal

dockertopia commented on engine/release-notes/index.md:
```diff
@@ -29,6 +29,15 @@ for Docker Engine.
 This release of Docker Engine comes with updated versions of the `compose`,
 `buildx`, `containerd`, and `runc` components, as well as some minor bugfixes.
```
```suggestion
`buildx`, `containerd`, and `runc` components, as well as some minor bug fixes.
```

dockertopia reviewed (commented):
``

thaJeztah commented on engine/release-notes/index.md:
```diff
@@ -29,6 +29,15 @@ for Docker Engine.
+> **Known issues**
+> 
+> We identified an issue with the [macOS CLI binaries](https://download.docker.com/mac/static/stable/)
+> in the 20.10.15 release. A fix for this issue [has been merged](https://github.com/docker/cli/pull/3592)
```
Ah, yes, I should do those (I know we haven't been doing for this document, but it's indeed the "correct" way to do it)

thaJeztah replied:
As to changing `fix for this issue has been merged` to `fix`; I generally try to make the title of the link slightly longer to help discoverability, and to make it easier to use (`fix` is very short for a link, and especially on mobile devices may be more tricky to "point" at)

thaJeztah commented:
Updated with your suggestions; changed the `fix` for the link to use `fix for this issue` (instead of the "has been merged"); PTAL if this looks good to you 🤗 

<img width="989" alt="Screenshot 2022-05-10 at 13 08 08" src="https://user-images.githubusercontent.com/1804568/167615487-be816b7e-4ed6-46d9-8c3a-572afcf00534.png">

usha-mandya commented:
Thanks @thaJeztah. Looks good. There's a broken link to the Extensions doc which isnt part of this PR. I've re-run the failed jobs to see if that fixes any issues.

thaJeztah commented:
Bleh.. looks to be still failing; broken link may be in the external repository though;

```
- ./_site/desktop/extensions-sdk/extensions/labels/index.html
Ran on 1666 files!


  *  linking to internal hash #design-principles that does not exist (line 197)
     <a href="../../design/design-overview/#design-principles">CSS guidlines</a>
```

thaJeztah commented:
Fix for the above is in https://github.com/docker/extensions-sdk/pull/160

usha-mandya reviewed (approved):
LGTM
